diff --git a/node_modules/react-native-bluetooth-classic/android/.classpath b/node_modules/react-native-bluetooth-classic/android/.classpath
deleted file mode 100644
index 32d6691..0000000
--- a/node_modules/react-native-bluetooth-classic/android/.classpath
+++ /dev/null
@@ -1,6 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<classpath>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-12/"/>
-	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
-	<classpathentry kind="output" path="bin/default"/>
-</classpath>
diff --git a/node_modules/react-native-bluetooth-classic/android/.project b/node_modules/react-native-bluetooth-classic/android/.project
index dcc6865..318f3e4 100644
--- a/node_modules/react-native-bluetooth-classic/android/.project
+++ b/node_modules/react-native-bluetooth-classic/android/.project
@@ -1,15 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>android</name>
+	<name>react-native-bluetooth-classic</name>
 	<comment>Project android created by Buildship.</comment>
 	<projects>
 	</projects>
 	<buildSpec>
-		<buildCommand>
-			<name>org.eclipse.jdt.core.javabuilder</name>
-			<arguments>
-			</arguments>
-		</buildCommand>
 		<buildCommand>
 			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
 			<arguments>
@@ -17,7 +12,6 @@
 		</buildCommand>
 	</buildSpec>
 	<natures>
-		<nature>org.eclipse.jdt.core.javanature</nature>
 		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
 	</natures>
 	<filteredResources>
diff --git a/node_modules/react-native-bluetooth-classic/android/.settings/org.eclipse.buildship.core.prefs b/node_modules/react-native-bluetooth-classic/android/.settings/org.eclipse.buildship.core.prefs
deleted file mode 100644
index f31e59e..0000000
--- a/node_modules/react-native-bluetooth-classic/android/.settings/org.eclipse.buildship.core.prefs
+++ /dev/null
@@ -1,13 +0,0 @@
-arguments=--init-script /var/folders/y3/j3s25xps7b7cxm467m3txhmh0000gn/T/db3b08fc4a9ef609cb16b96b200fa13e563f396e9bb1ed0905fdab7bc3bc513b.gradle --init-script /var/folders/y3/j3s25xps7b7cxm467m3txhmh0000gn/T/52cde0cfcf3e28b8b7510e992210d9614505e0911af0c190bd590d7158574963.gradle
-auto.sync=false
-build.scans.enabled=false
-connection.gradle.distribution=GRADLE_DISTRIBUTION(WRAPPER)
-connection.project.dir=
-eclipse.preferences.version=1
-gradle.user.home=
-java.home=/Library/Java/JavaVirtualMachines/jdk-11.0.14.jdk/Contents/Home
-jvm.arguments=
-offline.mode=false
-override.workspace.settings=true
-show.console.view=true
-show.executions.view=true
diff --git a/node_modules/react-native-bluetooth-classic/android/build.gradle b/node_modules/react-native-bluetooth-classic/android/build.gradle
index 0daf907..2b78a12 100644
--- a/node_modules/react-native-bluetooth-classic/android/build.gradle
+++ b/node_modules/react-native-bluetooth-classic/android/build.gradle
@@ -3,9 +3,8 @@ buildscript {
         google()
         mavenCentral()
     }
-
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.4.3'
+        classpath "com.android.tools.build:gradle:8.2.1"
     }
 }
 
@@ -16,12 +15,7 @@ def getExtOrDefault(name, fallback) {
 }
 
 android {
-    // Check AGP version for backward compatibility w/react-native versions still on gradle plugin 6
-    // it looks like this causes issues between 7 and 8, so for the time being 
-    def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION    
-    if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
-        namespace = "kjd.reactnative.bluetooth"
-    }
+    namespace "kjd.reactnative.bluetooth"
     
     compileSdkVersion getExtOrDefault('compileSdkVersion', 34)
     buildToolsVersion getExtOrDefault('buildToolsVersion', "34.0.0")
diff --git a/node_modules/react-native-bluetooth-classic/android/gradle.properties b/node_modules/react-native-bluetooth-classic/android/gradle.properties
new file mode 100644
index 0000000..646c51b
--- /dev/null
+++ b/node_modules/react-native-bluetooth-classic/android/gradle.properties
@@ -0,0 +1,2 @@
+android.useAndroidX=true
+android.enableJetifier=true
diff --git a/node_modules/react-native-bluetooth-classic/android/gradle/wrapper/gradle-wrapper.properties b/node_modules/react-native-bluetooth-classic/android/gradle/wrapper/gradle-wrapper.properties
index ffed3a2..3ae1e2f 100644
--- a/node_modules/react-native-bluetooth-classic/android/gradle/wrapper/gradle-wrapper.properties
+++ b/node_modules/react-native-bluetooth-classic/android/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.14.3-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
diff --git a/node_modules/react-native-bluetooth-classic/android/src/main/java/kjd/reactnative/bluetooth/conn/AbstractDeviceConnection.java b/node_modules/react-native-bluetooth-classic/android/src/main/java/kjd/reactnative/bluetooth/conn/AbstractDeviceConnection.java
index 42595a6..bdf06c9 100644
--- a/node_modules/react-native-bluetooth-classic/android/src/main/java/kjd/reactnative/bluetooth/conn/AbstractDeviceConnection.java
+++ b/node_modules/react-native-bluetooth-classic/android/src/main/java/kjd/reactnative/bluetooth/conn/AbstractDeviceConnection.java
@@ -104,6 +104,9 @@ abstract public class AbstractDeviceConnection implements DeviceConnection {
             mConnectionStatus = ConnectionStatus.CONNECTED;
             while (mConnectionStatus == ConnectionStatus.CONNECTED) {
                 bytes = mInStream.read(buffer);
+                if (bytes == -1) {
+                    break;
+                }
                 if (bytes > 0)
                     receivedData(Arrays.copyOf(buffer, bytes));
 
